### PR TITLE
Update scala to version 2.10.5

### PR DESCRIPTION
**Pull request by ajozwik**
See #1666, scala/scala#4186 solved problem with xml groups.

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

**Comment by eed3si9n** [comment]
Can you run scripted tests locally?

**Comment by eed3si9n** [comment]
The last Scala version bump happened on #1212. This didn't require change to scripted as far as I can tell.

**Comment by ajozwik** [comment]
I can rollback two last commits and you will see the result.
On my local machine first I call `sbt ";publishLocal;project bundledLauncherProj;publishLocal;project precompiled-2_9_2;publishLocal;project precompiled-2_8_2;publishLocal;project precompiled-2_9_3;publishLocal"` and then `sbt -Dsbt.version=0.13.9-SNAPSHOT <scripted name>`

**Comment by ajozwik** [comment]
Failed test - problem with bintray ...

My local result:

```
About to run tests: 
 * compiler-project/*

Running compiler-project / semantic-errors
Running compiler-project / inc-pickled-existential
Running compiler-project / macro-config
Running compiler-project / separate-analysis-per-scala
Running compiler-project / src-dep-plugin
Running compiler-project / error-in-invalidated
Running compiler-project / run2.8
....
```

File: .travis.yml [hunk 1]
```diff
@@ -42,7 +42,8 @@ notifications:
     - joshua.suereth@typesafe.com
 
 script:
-  - sbt -J-XX:ReservedCodeCacheSize=128m "$SCRIPTED_TEST"
+  - sbt ";publishLocal;project bundledLauncherProj;publishLocal;project precompiled-2_9_2;publishLocal;project precompiled-2_8_2;publishLocal;project precompiled-2_9_3;publishLocal"
```
Comment (eed3si9n) [inline .travis.yml:45]: If this is required, we should fix the build script instead of putting it in `.travis.yml` so all required JARs are published locally before the scripted tests.

File: .travis.yml [hunk 2]
```diff
@@ -42,7 +42,8 @@ notifications:
     - joshua.suereth@typesafe.com
 
 script:
-  - sbt -J-XX:ReservedCodeCacheSize=128m "$SCRIPTED_TEST"
+  - sbt ";publishLocal;project bundledLauncherProj;publishLocal;project precompiled-2_9_2;publishLocal;project precompiled-2_8_2;publishLocal;project precompiled-2_9_3;publishLocal"     
+  - sbt -J-XX:ReservedCodeCacheSize=128m -Dsbt.version=0.13.9-SNAPSHOT "$SCRIPTED_TEST"
```
Comment (eed3si9n) [inline .travis.yml:46]: Why is this needed all a sudden?

Comment (ajozwik) [inline .travis.yml:46]: The new process need to use sbt in version 0.13.9-SNAPSHOT. If we do not define the version - it will be red from `project/build.properties`. But there is 0.13.8. And we come back to the "NoSuchMethod" problem.

File: .travis.yml (hunk 1 above)
Comment (ajozwik) [inline .travis.yml:45]: It will be required only - when 0.13.9 will be published. After it the .travis script can be restored. 
Please checkout the code beaf39a196b530e335b44806069b3686dca43f38 and run the tests. You will receive: errors, see all the builds in: https://travis-ci.org/sbt/sbt/builds/54919087:

```
....
About to run tests:

* actions/*

java.lang.NoSuchMethodException: sbt.test.ScriptedTests$.run(java.io.File, boolean, [Ljava.lang.String;, java.io.File, [Ljava.lang.String;, scala.Function1)

at java.lang.Class.getMethod(Class.java:1667)

at Scripted$.doScripted(Scripted.scala:66)

at $0e28670edea040be68f9$$anonfun$scriptedTask$2$$anonfun$apply$2$$anonfun$apply$3.apply(build.sbt:474)

at $0e28670edea040be68f9$$anonfun$scriptedTask$2$$anonfun$apply$2$$anonfun$apply$3.apply(build.sbt:471)

at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
...
```

I guess it is problem with classloader. Mentioned in stack method exist.
As the solution the new process should be run - by scala 2.10.5 - and then all is run by the same classloader.

Comment (jsuereth) [inline .travis.yml:45]: Ah, I ran into this.  The way we use scripted is a bit gunky now.  We actually no longer have it isolated, so we need to fix it.

File: .travis.yml (hunk 2 above)
Comment (jsuereth) [inline .travis.yml:46]: @eed3si9n Scripted is borked in some way.  you can't change its code in the current repoa nd see those changes.  I suggest, maybe, we just remove it from the codebase and modularize it.

**Comment by jsuereth** [comment]
@eed3si9n We should figure out how to merge this before 0.13.9

**Comment by jsuereth** [comment]
Ok, so we bumped to 2.11.6, what should we do with this PR?

**Comment by ajozwik** [comment]
You can jest close without merging. 2.11.6 contains the same correction

**Comment by eed3si9n** [comment]
@jsuereth we bumped what to what?

**Comment by eed3si9n** [comment]
Closing this in favor of #2068
